tensorboard/webapp/plugins/npmi/views/annotations_list/header/header_test.ts

```diff
@@ -0,0 +1,154 @@
+import {By} from '@angular/platform-browser';
+
+import {Store, Action} from '@ngrx/store';
+import {provideMockStore, MockStore} from '@ngrx/store/testing';
+
+import {State} from '../../../../../app_state';
+import {HeaderComponent} from './header_component';
+import {HeaderContainer} from './header_container';
+import * as npmiActions from '../../../actions';
+import {SortingOrder} from '../../../store/npmi_types';
+import {appStateFromNpmiState, createNpmiState} from '../../../testing';
+import {getAnnotationSorting} from '../../../store';
+
+/** @typehack */ import * as _typeHackStore from '@ngrx/store';
+
+describe('Npmi Annotations List Header Container', () => {
+  let store: MockStore<State>;
+  let dispatchedActions: Action[];
+  let fixture: ComponentFixture<HeaderContainer>;
+  const css = {
+    CHECKBOX_CONTAINER: By.css('.toggle-all-container'),
+    ANNOTATIONS_HEADER_CONTAINER: By.css('.annotations-header-containers'),
+    CHECKBOX: By.css('mat-checkbox'),
+    HEADER: By.css('.header-clickable'),
+    SORT_ICON: By.css('.sorting-icon'),
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [HeaderComponent, HeaderContainer],
+      imports: [],
+      providers: [
+        provideMockStore({
+          initialState: appStateFromNpmiState(createNpmiState()),
+        }),
+      ],
+    }).compileComponents();
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+
+    dispatchedActions = [];
+    spyOn(store, 'dispatch').and.callFake((action: Action) => {
+      dispatchedActions.push(action);
+    });
+
+    fixture = TestBed.createComponent(HeaderContainer);
+    fixture.componentInstance.activeMetrics = [
+      'nPMI@test',
+      'nPMI@test2',
+      'nPMI_diff@(test-test2)',
+    ];
+    fixture.componentInstance.numAnnotations = 2;
+    fixture.componentInstance.annotations = {
+      annotation_1: [
+        {
+          annotation: 'annotation_1',
+          metric: 'test',
+          run: 'test',
+          nPMIValue: 1.0,
+          countValue: 100,
+        },
+      ],
+      annotation_2: [
+        {
+          annotation: 'annotation_2',
+          metric: 'test',
+          run: 'test',
+          nPMIValue: 1.0,
+          countValue: 100,
+        },
+      ],
+    };
+    fixture.detectChanges();
+  });
+
+  it('renders checkbox and container', () => {
+    const strippedMetrics = ['test', 'test2', '(test-test2)'];
+
+    const checkboxContainer = fixture.debugElement.query(
+      css.CHECKBOX_CONTAINER
+    );
+    expect(checkboxContainer).toBeTruthy();
+
+    const checkbox = fixture.debugElement.query(css.CHECKBOX);
+    expect(checkbox.nativeElement.checked).toBeFalsy();
+
+    const headerContainer = fixture.debugElement.query(
+      css.ANNOTATIONS_HEADER_CONTAINER
+    );
+    expect(headerContainer).toBeTruthy();
+
+    const headerMetrics = fixture.debugElement.queryAll(css.HEADER);
+    for (const index in strippedMetrics) {
```

Comment:
It's preferable to avoid control flow in tests, when feasible. This would be safer and more readable (from a testing perspective):

expect(headerMetrics[0].nativeElement.textContent.trim()).toBe('test');
expect(headerMetrics[1].nativeElement.textContent.trim()).toBe('test2')
expect(headerMetrics[2].nativeElement.textContent.trim()).toBe('(test-test2)');;